src/bundle/Command/MigrateNamespacesCommand.php

```diff
@@ -0,0 +1,164 @@
+            );
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $this->cursorStart = $input->getOption('cursor-start') !== null ? (int) $input->getOption('cursor-start') : null;
+        $this->cursorStop = $input->getOption('cursor-stop') !== null ? (int) $input->getOption('cursor-stop') : null;
+
+        // Check that both --cursor-start and cursor-start are set, or neither
+        if (($this->cursorStart === null) xor ($this->cursorStop === null)) {
+            throw new RuntimeException('The options --cursor-start and -cursor-stop are only for internal use !');
+        }
+
+        parent::execute($input, $output);
+
+        return self::SUCCESS;
+    }
+
+    protected function getObjectCount(): int
+    {
+        if ($this->objectCount === null) {
+            $this->output->writeln('Fetching number of objects to process. This might take several minutes if you have many records in ezcontentobject_attribute table.');
+            $this->objectCount = $this->gateway->countRichtextAttributes();
+        }
+
+        return $this->objectCount;
+    }
+
+    protected function iterate(): void
+    {
+        $limit = $this->getIterationCount();
+        $cursor = [
+            'start' => -1,
+            'stop' => null,
+        ];
+
+        $contentAttributeIDs = $this->gateway->getContentObjectAttributeIds($cursor['start'], $limit);
+        $cursor['stop'] = $this->getNextCursor($contentAttributeIDs);
+        while ($cursor['stop'] !== null) {
+            $this->createChildProcess($cursor, count($contentAttributeIDs));
+
+            $cursor['start'] = $cursor['stop'];
+            $contentAttributeIDs = $this->gateway->getContentObjectAttributeIds($cursor['start'], $limit);
+            $cursor['stop'] = $this->getNextCursor($contentAttributeIDs);
+        }
+    }
+
+    protected function completed(): void
+    {
+        $this->output->writeln(PHP_EOL . 'Completed');
+    }
+
+    protected function getNextCursor(array $contentAttributeIDs): ?int
+    {
+        $lastId = count($contentAttributeIDs) > 0 ? end($contentAttributeIDs)['id'] : null;
+
+        return $lastId;
+    }
+
+    protected function processData($cursor)
+    {
+        $this->updateNamespacesInColumns($cursor['start'], $cursor['stop']);
+    }
+
+    protected function constructCursorFromInputOptions(): mixed
+    {
+        return [
+            'start' => $this->cursorStart,
+            'stop' => $this->cursorStop,
+        ];
+    }
+
+    protected function addChildProcessArguments($cursor): array
+    {
+        return [
+            '--cursor-start=' . $cursor['start'],
+            '--cursor-stop=' . $cursor['stop'],
+        ];
+    }
+
+    protected function isChildProcess(): bool
+    {
+        return $this->cursorStart !== null || $this->cursorStop !== null;
+    }
+
+    public static function migrateNamespaces(string $xmlText)
```

Comment:
I put the migration code here, and not in a converter because our [converter interface](https://github.com/ibexa/fieldtype-richtext/blob/main/src/contracts/RichText/Converter.php#L25) expects a DOMDocument, and I don't want to waste time on converting back&forth to that, considering we might deal with millions of objects.

Reply:
If it is already a static method, couldn't it be placed in the converter? I find it a bit awkward that the converter calls a method on a symfony command and not the other way around.

Reply:
One reason for keeping it as-is, is that the converter is meant to be temporary. We'll not support on-the-fly conversion of namespaces forever. The command script will be kept though...